SignalServiceKit/src/Util/TypingIndicators.swift

```diff
@@ -0,0 +1,363 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc(OWSTypingIndicators)
+public protocol TypingIndicators: class {
+    // TODO: Use this method.
+    @objc
+    func didStartTypeOutgoingInput(inThread thread: TSThread)
+
+    // TODO: Use this method.
+    @objc
+    func didStopTypeOutgoingInput(inThread thread: TSThread)
```

Comment:
Done here - I had already made this change in another PR.

> Outgoing may be implied, but I'm fine with leaving it in for explicitness.

Yup, I was trying to avoid confusion.